core/pom.xml

```diff
@@ -181,6 +181,10 @@
       <groupId>commons-codec</groupId>
       <artifactId>commons-codec</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-compress</artifactId>
+    </dependency>
```

Comment:
<img width="928" alt="image" src="https://user-images.githubusercontent.com/15246973/210139366-9ea00e18-8c93-446a-b98f-8f0162887abf.png">
https://github.com/panbingkun/spark/actions/runs/3811706860/jobs/6485303553

Reply:
Use `CountingOutputStream` class, so direct dependency should be added
https://github.com/apache/spark/blob/5300028a6fcab2ef118b4d96b2a4186542c030d5/core/src/main/scala/org/apache/spark/deploy/history/EventLogFileWriters.scala#L23-L24

Reply:
Do you know why this is required by 1.22 only?

Reply:
I will investigate root cause.

Reply:
If this isn't just a breaking change in commons-compress, then I feel like we have two different versions of common-compress in play, which could explain why just adding the dependency explicitly changed behavior